### ISSUE TITLE
Babies don't get fed at all when replimat and Vanilla Animals Expanded: Caves is installed.

### ISSUE THREAD

**Issue by ruifung** [sumghai/Replimat#73]
**Describe the bug:**  

When both Vanilla Animals Expanded: Caves and replimat is installed, all attempts to feed a baby without a mother will throw an error.

**Expected behavior:**  

The baby to be fed from either a replimat terminal or baby food in storage.

**Steps to reproduce:**  

1. Install both replimat and Vanilla Animals Expanded: Caves (And it's pre-requisites)
2. Spawn a baby pawn
3. Set it's food need to low.
4. Watch when a pawn attempt to feed it an error out.

Alternate steps:
1. Try to tell a pawn to feed the baby.
2. The right click menu doesn't come out and an error is thrown.

**Additional information:**

```
Exception in BreadthFirstTraverse: System.NullReferenceException: Object reference not set to an instance of an object
  at RimWorld.FoodUtility.WillEat (Verse.Pawn p, Verse.ThingDef food, Verse.Pawn getter, System.Boolean careIfNotAcceptableForTitle) [0x0000d] in <1ea541f48818430abd724fc665f76dc4>:0 
  at Replimat.ReplimatUtility.RepMatWillEat (Verse.Pawn p, Verse.ThingDef food, Verse.Pawn getter) [0x00008] in <a38138ea29294fd19de33371a37a261a>:0 
  at Replimat.ReplimatUtility+<>c__DisplayClass8_0.<PickMeal>b__0 (Verse.ThingDef x) [0x00000] in <a38138ea29294fd19de33371a37a261a>:0 
  at System.Linq.Enumerable+WhereEnumerableIterator`1[TSource].ToList () [0x0001b] in <351e49e2a5bf4fd6beabb458ce2255f3>:0 
  at System.Linq.Enumerable.ToList[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x0001f] in <351e49e2a5bf4fd6beabb458ce2255f3>:0 
  at Replimat.ReplimatUtility.PickMeal (Verse.Pawn eater, Verse.Pawn getter) [0x00061] in <a38138ea29294fd19de33371a37a261a>:0 
  at Replimat.Harmony_FoodUtility_GetFinalIngestibleDef.Prefix (Verse.Thing& foodSource, Verse.ThingDef& __result) [0x00010] in <a38138ea29294fd19de33371a37a261a>:0 
  at (wrapper dynamic-method) RimWorld.FoodUtility.RimWorld.FoodUtility.GetFinalIngestibleDef_Patch1(Verse.Thing,bool)
  at (wrapper dynamic-method) RimWorld.FoodUtility.RimWorld.FoodUtility.SpawnedFoodSearchInnerScan_Patch1(Verse.Pawn,Verse.IntVec3,System.Collections.Generic.List`1<Verse.Thing>,Verse.AI.PathEndMode,Verse.TraverseParms,single,System.Predicate`1<Verse.Thing>)
  at (wrapper dynamic-method) RimWorld.FoodUtility.RimWorld.FoodUtility.BestFoodSourceOnMap_Patch2(Verse.Pawn,Verse.Pawn,bool,Verse.ThingDef&,RimWorld.FoodPreferability,bool,bool,bool,bool,bool,bool,bool,bool,bool,bool,bool,RimWorld.FoodPreferability,System.Nullable`1<single>)
  at RimWorld.FoodUtility.TryFindBestFoodSourceFor (Verse.Pawn getter, Verse.Pawn eater, System.Boolean desperate, Verse.Thing& foodSource, Verse.ThingDef& foodDef, System.Boolean canRefillDispenser, System.Boolean canUseInventory, System.Boolean canUsePackAnimalInventory, System.Boolean allowForbidden, System.Boolean allowCorpse, System.Boolean allowSociallyImproper, System.Boolean allowHarvest, System.Boolean forceScanWholeMap, System.Boolean ignoreReservations, System.Boolean calculateWantedStackCount, RimWorld.FoodPreferability minPrefOverride) [0x000ff] in <1ea541f48818430abd724fc665f76dc4>:0 
  at RimWorld.ChildcareUtility.FindBabyFoodForBaby (Verse.Pawn feeder, Verse.Pawn baby) [0x00012] in <1ea541f48818430abd724fc665f76dc4>:0 
  at RimWorld.WorkGiver_BottleFeedBaby.JobOnThing (Verse.Pawn pawn, Verse.Thing t, System.Boolean forced) [0x00014] in <1ea541f48818430abd724fc665f76dc4>:0 
  at RimWorld.WorkGiver_Scanner.HasJobOnThing (Verse.Pawn pawn, Verse.Thing t, System.Boolean forced) [0x00000] in <1ea541f48818430abd724fc665f76dc4>:0 
  at RimWorld.JobGiver_Work+<>c__DisplayClass3_1.<TryIssueJobPackage>b__0 (Verse.Thing t) [0x00013] in <1ea541f48818430abd724fc665f76dc4>:0 
  at Verse.RegionProcessorClosestThingReachable.RegionProcessor (Verse.Region reg) [0x000f2] in <1ea541f48818430abd724fc665f76dc4>:0 
  at Verse.RegionTraverser+BFSWorker.BreadthFirstTraverseWork (Verse.Region root, Verse.RegionEntryPredicate entryCondition, Verse.RegionProcessor regionProcessor, System.Int32 maxRegions, Verse.RegionType traversableRegionTypes) [0x00054] in <1ea541f48818430abd724fc665f76dc4>:0 
  at Verse.RegionTraverser.BreadthFirstTraverse (Verse.Region root, Verse.RegionEntryPredicate entryCondition, Verse.RegionProcessor regionProcessor, System.Int32 maxRegions, Verse.RegionType traversableRegionTypes) [0x00044] in <1ea541f48818430abd724fc665f76dc4>:0 
UnityEngine.StackTraceUtility:ExtractStackTrace ()
Verse.Log:Error (string)
Verse.RegionTraverser:BreadthFirstTraverse (Verse.Region,Verse.RegionEntryPredicate,Verse.RegionProcessor,int,Verse.RegionType)
Verse.RegionTraverser:BreadthFirstTraverse (Verse.Region,Verse.RegionProcessorDelegateCache,int,Verse.RegionType)
Verse.GenClosest:RegionwiseBFSWorker (Verse.IntVec3,Verse.Map,Verse.ThingRequest,Verse.AI.PathEndMode,Verse.TraverseParms,System.Predicate`1<Verse.Thing>,System.Func`2<Verse.Thing, single>,int,int,single,int&,Verse.RegionType,bool)
Verse.GenClosest:ClosestThingReachable (Verse.IntVec3,Verse.Map,Verse.ThingRequest,Verse.AI.PathEndMode,Verse.TraverseParms,single,System.Predicate`1<Verse.Thing>,System.Collections.Generic.IEnumerable`1<Verse.Thing>,int,int,bool,Verse.RegionType,bool)
RimWorld.JobGiver_Work:TryIssueJobPackage (Verse.Pawn,Verse.AI.JobIssueParams)
Verse.AI.ThinkNode_PrioritySorter:TryIssueJobPackage (Verse.Pawn,Verse.AI.JobIssueParams)
Verse.AI.ThinkNode_Priority:TryIssueJobPackage (Verse.Pawn,Verse.AI.JobIssueParams)
Verse.AI.ThinkNode_Tagger:TryIssueJobPackage (Verse.Pawn,Verse.AI.JobIssueParams)
Verse.AI.ThinkNode_Subtree:TryIssueJobPackage (Verse.Pawn,Verse.AI.JobIssueParams)
Verse.AI.ThinkNode_Priority:TryIssueJobPackage (Verse.Pawn,Verse.AI.JobIssueParams)
Verse.AI.ThinkNode_Conditional:TryIssueJobPackage (Verse.Pawn,Verse.AI.JobIssueParams)
Verse.AI.ThinkNode_Priority:TryIssueJobPackage (Verse.Pawn,Verse.AI.JobIssueParams)
Verse.AI.Pawn_JobTracker:DetermineNextJob (Verse.ThinkTreeDef&)
Verse.AI.Pawn_JobTracker:TryFindAndStartJob ()
Verse.AI.Pawn_JobTracker:EndCurrentJob (Verse.AI.JobCondition,bool,bool)
Verse.AI.Pawn_JobTracker:JobTrackerTick ()
Verse.Pawn:Tick ()
Verse.TickList:Tick ()
(wrapper dynamic-method) Verse.TickManager:Verse.TickManager.DoSingleTick_Patch2 (Verse.TickManager)
Verse.TickManager:TickManagerUpdate ()
Verse.Game:UpdatePlay ()
Verse.Root_Play:Update ()
```

## System and Game Configuration
**Operating System:** Windows 11 64-bit
**Physical RAM:** 32GB
**RimWorld base game version:** 1.4.3542
**RimWorld DLCs loaded:** Royalty, Ideology, Biotech
**Replimat mod version:** 1.4.2
**Replimat mod downloaded from:** Steam
**Your mod list:** 
- Harmony - 2.2.2
- Core [https:// no link (local mod)]
- Royalty [Official DLC] [https:// no link (local mod)]
- Ideology [Official DLC] [https:// no link (local mod)]
- Biotech [Official DLC] [https:// no link (local mod)]
- Replimat - 1.4.2
- Vanilla Expanded Framework - Can't find version number
- Vanilla Animals Expanded - Caves v2.02

## Declaration

I hereby verify that:
- [x] I am running the latest version of the game, Replimat and its dependencies
- [x] I am NOT running any mods with known incompatibilities with Replimat 
- [x] I have narrowed down my modlist to the fewest mods needed to cause this issue
- [x] I have confirmed that the issue does not happen if Replimat is disabled
- [x] I have searched the issue tracker and confirmed no-one else has reported my issue before
- [x] I understand that my issue will be closed and ignored if I do not fill this form properly

**Comment by ruifung** [comment]
Addendum: If *either* VAE:Caves or Replimat is disabled, babies get fed just fine.

**Comment by sumghai** [comment]
I've reproduced the bug in my dev environment, but the root cause is _not_ Replimat.

Firstly, the log message states that the NullRef is being thrown from `RimWorld.FoodUtility.WillEat()`, which is a base game function. Replimat is simply calling it to get a boolean flag for whether any given food item can be eaten by a particular pawn.

After adding some extra debugging statements to Replimat, I found that `VAECaves_EggSalamanderFertilized` is missing an `ingestible` node, which breaks `RimWorld.FoodUtility.WillEat()` because the method is looking for `ingestible.babiesCanIngest` but obviously cannot find it. Commenting out the salamander egg allows babies to be fed from Replimat Terminals again.

Examining `VAECaves_EggSalamanderFertilized` more closely, it appears that (based on the item description) salamander eggs are supposed to be edible, but currently aren't.

This will need to be fixed by the VE team.
